pkgs/applications/graphics/vengi-tools/default.nix

```diff
@@ -58,9 +58,15 @@ stdenv.mkDerivation rec {
     #postgresql
     #libpqxx
     #mosquitto
-  ] ++ lib.optional stdenv.isLinux wayland-protocols
-    ++ lib.optionals stdenv.isDarwin [ Carbon CoreServices OpenCL ]
-    ++ lib.optional (!stdenv.isDarwin) opencl-headers;
+  ] ++ lib.optionals stdenv.isLinux [
+    wayland-protocols
```

Comment:
@drupol why did you change `optional` to single-element `optionals`?

Reply:
Hi!

For many reasons:

1. Since `lib.optional` does more than expected. It wraps its contents into a list.
2. Some people consider it harmful: https://discourse.nixos.org/t/lib-optional-considered-harmful/7081
3. Since we are dealing with `buildInputs` which is a list containing potential more than 1 element, using the plural form `lib.optionals` seems more natural.
4. I've been submitting a couple of PRs against `nixpkgs` before and some maintainers asked me to use `lib.optionals` instead of `lib.optional`.

Reply:
> 1. Since `lib.optional` does more than expected. It wraps its contents into a list.

No, that's entirely expected. The docstring says: "Return a singleton list or an empty list, depending on a boolean value"

> 2. Some people consider it harmful: https://discourse.nixos.org/t/lib-optional-considered-harmful/7081

https://discourse.nixos.org/t/lib-optional-considered-harmful/7081/5

> 3. Since we are dealing with `buildInputs` which is a list containing potential more than 1 element, using the plural form `lib.optionals` seems more natural.

As you wrote, `lib.optional` _does_ return a list, so this seems like personal preference. There are no recommendations and `lib.optional` is used widely. In this case I think `optional` makes the expression more legible since it's less verbose.

> 4. I've been submitting a couple of PRs against `nixpkgs` before and some maintainers asked me to use `lib.optionals` instead of `lib.optional`.

I can somewhat understand it on new packages, but it doesn't make sense to do the change now, it's just diff noise. We can change it when and if more optional dependencies are added.